### PR TITLE
fix: error handling for trade desk first party data flow

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?

Improve error handling for trade desk first party data flow.
- Added invalid action type check (Valid action types :- insert/delete)
- Empty 'fields' check

## What is the related Linear task?

Resolves INT-1378

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/cdk/v2/destinations/the_trade_desk/utils.js [hunk 1]
```diff
@@ -59,13 +59,18 @@ const processRecordInputs = (inputs, destination) => {
...
+  const emptyFieldsError = new InstrumentationError('Fields cannot be empty');
 
   inputs.forEach((input) => {
     const { fields, action } = input.message;
     const isInsertOrDelete = action === 'insert' || action === 'delete';
 
-    if (isInsertOrDelete) {
+    if (!isInsertOrDelete) {
+      errorResponseList.push(handleRtTfSingleEventError(input, invalidActionTypeError, {}));
+    } else if (!fields || Object.keys(fields).length === 0) {
```
Comment (koladilip) [inline src/cdk/v2/destinations/the_trade_desk/utils.js:71]: don't have util for isEmptyObject?

Comment (Gauravudia) [reply src/cdk/v2/destinations/the_trade_desk/utils.js]: We have it, Updated the code.

File: src/cdk/v2/destinations/the_trade_desk/utils.js [hunk 2]
```diff
@@ -59,13 +60,18 @@ const processRecordInputs = (inputs, destination) => {
...
-  const error = new InstrumentationError('Invalid action type');
+  const invalidActionTypeError = new InstrumentationError('Invalid action type');
+  const emptyFieldsError = new InstrumentationError('Fields cannot be empty');
 
   inputs.forEach((input) => {
     const { fields, action } = input.message;
     const isInsertOrDelete = action === 'insert' || action === 'delete';
 
-    if (isInsertOrDelete) {
+    if (!isInsertOrDelete) {
```
Comment (koladilip) [inline src/cdk/v2/destinations/the_trade_desk/utils.js:70]: we can write this like this:
if (!isInsertOrDelete) {
    errorResponseList.push(handleRtTfSingleEventError(input, invalidActionTypeError, {}));
    return;
}

if (isEmptyObject(fields)) {
      errorResponseList.push(handleRtTfSingleEventError(input, emptyFieldsError, {}));
      return;
    }
    
    this bit more readable

Comment (Gauravudia) [reply src/cdk/v2/destinations/the_trade_desk/utils.js:70]: Agreed!!
Incorporate the suggestions.

**Review by koladilip — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**